pkg/controller/registrysyncer/registrysyncer.go

```diff
@@ -176,7 +176,7 @@ func (r *reconciler) reconcile(ctx context.Context, req reconcile.Request, log *
 			}
 		}
 
-		if err := r.ensureImageStream(ctx, sourceImageStream, client, log); err != nil {
+		if err := r.ensureImageStream(ctx, sourceImageStream, client, log); err != nil && !apierrors.IsAlreadyExists(err) {
```

Comment:
The underlying code is specifically built to support an upsert. If you are getting an IsAlreadyExists, you are using the `CreateOrUpdate` wrong.
/hold

Reply:
As discussed in slack, added a blocking get to confirm the appearance in cache.

/hold cancel